src/CoreBundle/private/library/classes/TCMSURLHistory.class.php

```diff
@@ -21,7 +21,35 @@ class TCMSURLHistory
      */
     public $aHistory = array();
 
-    public $index = 0;
+    public function __get($name)
+    {
+        if ('index' === $name) {
+            @trigger_error('The property TCMSURLHistory::$index is deprecated.', E_USER_DEPRECATED);
+
+            return $this->getHistoryCount();
+        }
+
+        $trace = debug_backtrace();
+        trigger_error(sprintf('Undefined property via __get(): %s in %s on line %s',
+            $name,
+            $trace[0]['file'],
+            $trace[0]['line']),
+            E_USER_NOTICE);
+
+        return null;
+    }
+
+    public function __set($name, $val)
+    {
+        if ('index' === $name) {
+            throw new \LogicException('Invalid property: '.$name);
```

Comment:
Can we trigger a deprecation notice here? And a "plain" notice for everything else? I think this is closest to PHP default behavior for nonexisting properties.

Reply:
done